content/articles/nodejs-mongoosejs-mongodb/index.md

```diff
@@ -0,0 +1,211 @@
+### How to Connect MongoDB to Node.js Using Mongoose
+MongoDB is one of the most widely used No-SQL databases in the developer world today. No-SQL databases allow developers to send and retrieve data as JSON documents, instead of SQL objects. To work with MongoDB in a Node.js app, we can use Mongoose.
+
+[Mongoose.js](https://mongoosejs.com) connects your MongoDB clusters or collections with your Node.js app. It enables you to create schemas for your documents. Mongoose provides a lot of functionality for creating and working with schemas. 
+
+In this tutorial we will look at how to connect a MongoDB instance with a Node.js application.
+
+### Prerequisites
+Before we move on, you'll need to have the following:
+- Node.js installed on your machine.
+- A MongoDB instance running on your machine. You won't need this if you want to use MongoDB Atlas.
+- Some knowledge of Node.js and Express.js.
+
+### Step 1 - Installing Mongoose on a Node.js Environment
+Create and navigate to a new folder by running the following commands on a terminal.
+
+```bash
+$ mkdir mongoose_tutorial
+$ cd mongoose_tutorial
+```
+
+Then install Express and Mongoose by executing the following command on a terminal.
+
+```bash
+$ npm install express mongoose --save
+```
+
+If you are using Yarn, run:
+
+```bash
+$ yarn add express mongoose
+```
+
+### Step 2 -- Creating the connection
+Create a new file `server.js` for starting our Express.js server. Load `mongoose` and `express` by adding the following code to `server.js`.
+
+`server.js`
+```javascript
+const express = require("express");
+const mongoose = require("mongoose");
+const Router = require("./routes")
+
+const app = express();
+
+app.use(express.json());
+```
+
+Then connect to a local MongoDB instance using the `mongoose.connect()` function.
+
+`server.js`
+```javascript
+mongoose.connect('mongodb://localhost:27017/usersdb',
+  {
+    useNewUrlParser: true,
+    useFindAndModify: false,
+    useUnifiedTopology: true
+  }
+);
+```
+
+We pass the `useNewUrlParser: true`, etc.  to `mongoose.connect()` to avoid the [`DeprecationWarning`](https://mongoosejs.com/docs/deprecations.html).
+
+To create a connection to MongoDB Atlas, use the following code. Replace the variables using your actual credentials.
+
+`server.js`
+```javascript
+const username = "<mongodb username>";
+const password = "<password>";
+const cluster = "<cluster name>";
+const dbname = "myFirstDatabase";
+
+mongoose.connect(
+  `mongodb+srv://${username}:${password}@${cluster}.mongodb.net/${dbname}?retryWrites=true&w=majority`,
```

Comment:
This connection string did not work for me.
I'm guessing this might be the case for a few of the people who might try to follow the article.

Here is a breakdown of how I went about getting a connection string that worked. Consider adding these steps to help readers figure out how to do the same:

1. I navigated to the Clusters tab on MongoDb Atlas

![image](https://user-images.githubusercontent.com/47227271/120064494-1884e000-c075-11eb-82af-5d4f6021d846.png)

2. Clicked on the connect button found under my cluster name "PBP".

![image](https://user-images.githubusercontent.com/47227271/120064518-381c0880-c075-11eb-8170-d464b82d0334.png)

3. Clicked on the "Connect your application option"

![image](https://user-images.githubusercontent.com/47227271/120064551-6994d400-c075-11eb-813a-366bd4cb3d37.png)

4. I then copied the connection string provided:

![image](https://user-images.githubusercontent.com/47227271/120064589-a3fe7100-c075-11eb-9b17-944d004565f2.png)

5. Finally I modified the string to match the parameters needed for the connection in the App:

```JavaScript
  mongoose.connect(
    `mongodb+srv://${username}:${password}@${cluster}.dkuv8.azure.mongodb.net/${dbname}?retryWrites=true&w=majority`,
  {
    useNewUrlParser: true,
    useFindAndModify: false,
    useUnifiedTopology: true,
  }
);
```

Reply:
Also consider adding a way to check if the connection worked. To help the reader figure out if their connection was successful or not.

I used the following lines of code to achieve that for myself :

```JavaScript
const db = mongoose.connection;
db.on("error", console.error.bind(console, "connection error:"));
db.once("open", function () {
  console.log("Connected successfully");
});
```

Reply:
Hello @adrianmurage 

The string didn't work for you because the cluster variable included `.dkuv8.azure`. This part is included in the variable because this part of the connection string changes per cluster.

I'm sorry I hadn't mentioned this.